### DISCUSSION TITLE
Tipo de Datos Boolean

### DISCUSSION THREAD

**Discussion by earivas** [tmijail/Odoo-Power-BI-Connector#26] [Q&A] [answered]
De antemano quisiera agradecer por el conector y por el excelente trabajo.  

Quisiera saber a que se debe el siguiente caso, en odoo16 cuando se archiva un producto, en el modelo product.product el campo active pasa del estado TRUE a FALSE  como lo podemos observar en la siguiente imagen (script realizado directamente en postgresql)

![image](https://github.com/tmijail/Odoo-Power-BI-Connector/assets/11370622/336c0cc5-6e61-4677-89a4-cc0df70dbbae)

Pero al momento de importar los datos con Power Query mediante el conector para Power BI, el registro con el id 37 que corresponde al producto archivado, no se muestra a pesar de que no existe ningún filtro 

![image](https://github.com/tmijail/Odoo-Power-BI-Connector/assets/11370622/bfc12f72-71ac-452d-98b3-3304f0a9bffc)

Si la importación la realizo directamente de la base de datos, los resultados son esperados

 
![image](https://github.com/tmijail/Odoo-Power-BI-Connector/assets/11370622/9e9cd028-004a-4719-88a5-d17a1711a54e)

Si regreso a Odoo y vuelvo a activar el producto 37 cuyo estado vuelve a TRUE, como se ve en la siguiente imagen  

![image](https://github.com/tmijail/Odoo-Power-BI-Connector/assets/11370622/25213a5c-3dcf-45c0-9c72-9c0d7f34ef24)

Y Procedo a actualizar los datos en Power Query, inmediatamente se muestra el articulo con el id 37 cuando su estado es TRUE. 

![image](https://github.com/tmijail/Odoo-Power-BI-Connector/assets/11370622/873d10a1-f6e1-447f-a8f8-9d7cfc6a0cd9)

¿Sabes que puedo estar haciendo mal o a que se debe ese comportamiento?

Para aclarar que la lectura de los datos, no se debe realizar directamente de la base de datos, sino por medio del conector.
Muy agradecido por la ayuda que me puedan brindar

**Comment by tmijail** [comment, marked as answer]
Cómo pudiste comprobar, cuando se consultan los datos por la API, por defecto no se devuelven los registros archivados. Ese comportamiento se puede cambiar usando el contexto `active_test`.

Por ejemplo, para buscar todos los `product.template` (incluso los archivados) se usa `search_read` de la siguiente manera:

```
=search_read('product.template', {}, [context = [active_test = False]])
```

**Comment by earivas** [comment]
Mil y mil gracias, funciona perfecto, en mi caso funciono con **false** en lugar de **False**... muy agradecido

Saludos
